File: ruby/004_leap_years/code.rb
# si es divisible entre 4, excepto aquellos divisibles entre 100 pero no entre 400.
number = ARGV.empty? ? Time.now.year : ARGV[0].to_i
div = (number % 400).zero? || !(number % 100).zero?
if (number % 4).zero? && div
  puts 'It is'
else
  puts 'It isn\'t'
end

File: python/004_leap_years/code.py
import sys
import datetime

sys.argv.pop(0)
number = int(sys.argv[0]) if sys.argv else datetime.datetime.now().year

if number % 4 == 0 & (number % 100 != 0 | number % 400 == 0):
    print('It is')
else:
    print('It isn\'t')
